Functions/Quantification/xASL_quant_BSupCalculation.m

```diff
@@ -98,14 +98,15 @@
  
 % First phase calculation for a graph visual output
 if ~isempty(PathGraph)
+	% This part is used for visualization only and not for calculation the signal reduction
     % Preallocate (sum is required if multiPLD and therefore multiple ReadoutTime is present)
 	SignalPercentageVector = zeros(round(ReadoutTime + max(SliceTime)), 1);
 	
 	if PresaturationTime
-		SignalPercentageVector(1:PresaturationTime,1) = 1;
-		SignalPercentageVector(PresaturationTime+1:BackgroundSuppressionPulseTime(1),1) = 1 - (1 - 0)*exp(-(1:(BackgroundSuppressionPulseTime(1)-PresaturationTime))/T1Time);
+		SignalPercentageVector(1:round(PresaturationTime), 1) = 1;
+		SignalPercentageVector(round(PresaturationTime+1):round(BackgroundSuppressionPulseTime(1)), 1) = 1 - (1 - 0)*exp(-(1:round(BackgroundSuppressionPulseTime(1)-PresaturationTime))/T1Time);
```

Comment:
These dimensions are different. Left side is N rows x 1 column, right side is 1 row x N columns.